### ISSUE TITLE
`CheckboxGroup` has `aria-conditional-attr` violation

### ISSUE THREAD

**Issue by joshblack** [primer/react#4262]
### Description

The `CheckboxGroup` and `Checkbox` components have an [`aria-conditional-attr` violation](https://dequeuniversity.com/rules/axe/4.8/aria-conditional-attr?application=AxeChrome) from axe due to usage of `aria-checked`

The value of `aria-checked` should mirror the value of the input's `checked` property

### Steps to reproduce

1. Visit the [component storybook](https://primer.style/react/storybook/iframe.html?args=&id=components-checkboxgroup--default&viewMode=story)
2. Run axe
3. View issue titled: "ARIA attributes must be used as specified for the element's role"

### Version

v36.x
